cmd/helm/install.go

```diff
@@ -211,6 +214,7 @@ func newInstallCmd(c helm.Interface, out io.Writer) *cobra.Command {
 	f.StringVar(&inst.version, "version", "", "specify the exact chart version to install. If this is not specified, the latest version is installed")
 	f.Int64Var(&inst.timeout, "timeout", 300, "time in seconds to wait for any individual Kubernetes operation (like Jobs for hooks)")
 	f.BoolVar(&inst.wait, "wait", false, "if set, will wait until all Pods, PVCs, Services, and minimum number of Pods of a Deployment are in a ready state before marking the release as successful. It will wait for as long as --timeout")
+	f.BoolVar(&inst.safe, "safe", false, "if set, upgrade process rolls back changes made in case of failed upgrade")
```

Comment:
Maybe better something like `rollback-on-error` instead of `safe`?

Reply:
I think this is probably a good idea. For `upgrade` you'd have `--rollback-on-error` and for `install` you'd have `--purge-on-error`. The only worry I have is that the flag name is different for each one and you have to know what is going on underneath to understand why they are different for each command

Reply:
Just brainstorming here with a few more names:

`--cleanup-on-error`
`--auto-recover`

Any thoughts on this @Bessonov and @cleac? No need to really bike shed too much here, but we do want to have clear flag names

Reply:
I'm fine with every one, but `safe`. This change execute `delete` with `purge=true` on error. May be then `--purge-on-error` is more clear.

Reply:
I think I agree. the `*-on-error` flags don't leave much room for misinterpretation

Reply:
From my point of view, `atomic` is misleading too. For me, `atomic` is like transactional for a database. But this implementation does it in two step, not one.

Reply:
@cleac Let's go with `--atomic` then. It is concise enough to not be overly verbose and should be clear enough, especially with documentation on the flag